### PR TITLE
Stratos from SUSE Helm release

### PR DESCRIPTION
**Pull request by f0rmiga**
## Description

- Adds the SUSE Helm repo to the Vagrant machine.
- Refactors the Stratos deployment to use SUSE Helm repo.
- Refactors environment variables to defaults.
- Fixes minor things.

## Test plan

- Start a Vagrant machine.
- Deploy SCF and UAA with `make run`.
- Deploy Stratos using `make stratos-run` or `make/stratos/run`.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -197,7 +197,7 @@ Vagrant.configure(2) do |config|
 
     if [ -d "#{mounted_custom_setup_scripts}/provision.d" ]; then
       scripts=($(find "#{mounted_custom_setup_scripts}/provision.d" -iname "*.sh" -executable -print | sort))
-      for script in ${scripts}; do
+      for script in ${scripts[@]}; do
```
Comment (mook-as) [inline Vagrantfile]: Quote? `"${scripts[@]}"` (though it doesn't actually change anything)

File: make/include/defaults [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+set -o errexit -o nounset
+
+: "${GIT_ROOT:=$(git rev-parse --show-toplevel)}"
+source "${GIT_ROOT}/.envrc"
+
+: "${CF_NAMESPACE:=cf}"
+: "${DOMAIN:=cf-dev.io}"
+: "${UAA_CF_IDENTITY_ZONE:=cf}"
```
Comment (mook-as) [inline make/include/defaults]: Isn't this always the CF namespace, as set in [uaa-create-user](https://github.com/SUSE/scf/blob/d4858f0d424fbd008f34d9a9ef3e04ad9256cc1f/src/scf-release/jobs/uaa-create-user/templates/run.erb#L69)?

File: make/include/defaults [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+set -o errexit -o nounset
+
+: "${GIT_ROOT:=$(git rev-parse --show-toplevel)}"
+source "${GIT_ROOT}/.envrc"
+
+: "${CF_NAMESPACE:=cf}"
```
Comment (mook-as) [inline make/include/defaults]: This should match up to where we [deploy SCF](https://github.com/SUSE/scf/blob/d4858f0d424fbd008f34d9a9ef3e04ad9256cc1f/make/run#L10)

Comment (f0rmiga) [reply make/include/defaults]: The refactor of UAA and SCF will come on a separate PR.

File: make/include/defaults (hunk 2 above)
Comment (f0rmiga) [reply make/include/defaults]: Cool, thanks for pointing out. I wasn't sure if they always matched.

File: make/include/defaults [hunk 4]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+set -o errexit -o nounset
+
+: "${GIT_ROOT:=$(git rev-parse --show-toplevel)}"
+source "${GIT_ROOT}/.envrc"
+
+: "${NAMESPACE:=cf}"
+: "${CF_NAMESPACE:=${NAMESPACE}}"
```
Comment (mook-as) [inline make/include/defaults]: I don't think we can actually override `CF_NAMESPACE` to something that isn't `${NAMESPACE}`?

**Comment by f0rmiga** [comment]
[(5) Include Stratos in Vagrant to measure resource metrics](https://trello.com/c/ZyqXBasA/891-5-include-stratos-in-vagrant-to-measure-resource-metrics)

File: make/include/defaults [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+: "${CF_CHART:="${GIT_ROOT}/output/helm"}"
+: "${DOMAIN:=cf-dev.io}"
+: "${UAA_NAMESPACE:=uaa}"
+: "${UAA_HOST:=uaa.${DOMAIN}}"
+: "${UAA_PORT:=2793}"
+: "${UAA_CLIENT_ID:=cf}"
```
Comment (mook-as) [inline make/include/defaults]: And `UAA_CLIENT_ID` has to be `cf`, due to the [hard-coded client id](https://github.com/SUSE/scf/blob/04f0c59c6d55151d2a440c11e9508d16f424d668/container-host-files/etc/scf/config/opinions-common.yml#L431)?

**Pull request merged**